scripts/sct_compute_mtsat.py

```diff
@@ -132,9 +127,49 @@ def get_parser():
     return parser
 
 
-def main():
+def get_tr_and_flipangle(args):
+    """
+    Retrieve metadata from the json sidecar or from the input arguments.
+    :param args:
+    :return: trmt, trpd, trt1, famt, fapd, fat1
+    """
+    # TODO: deal with the code duplication below
```

Comment:
To simplify this code you can leverage the `default` param of `add_argument()` and avoid checking the optional arguments.

For example:
```
mandatoryArguments.add_argument(
        "-trmt",
        help="TR [in ms] for mt image. By default, will be fetch from the json sidecar (if it exists).",
        type=float,
        default=fetch_metadata(get_json_file_name(args.mt, check_exist=True), 'RepetitionTime')
        metavar=Metavar.float,
        )
```

Since all args checked in `get_tr_and_flipangle()` are optional, you can do this and get rid of the `get_tr_and_flipangle()` function all together.

Reply:
that sounds like a genius idea indeed, but i'm having an issue implementing it because `args.mt` is not known yet. Before i spend too much time finding a solution, maybe you know what to do?

Reply:
Maybe if the order of the arguments is changed so that `args.mt` is before other arguments that try to use it? If it makes sense for the usage interface of course.

Reply:
it is currently placed before, but within the `get_parser()` function, which does not know about the `args` variable (this variable is generated by `parser.parse_args` inside the main()

Reply:
Ok we can optimize later, not critical.

Reply:
i think i found a workaround...

Reply:
here: d953ad37eb7470d43d81f123e134729902c5724e
the trick was to use `parse_known_args()` (ignore arguments that are not defined in the parser yet)

Reply:
Good to know!

Reply:
trickier than that... now the unit test is failing, because it cannot get the arguments passed to main(arguments):
~~~
usage: sct_compute_mtsat -mt <file> -pd <file> -t1 <file>
sct_compute_mtsat: error: the following arguments are required: -mt, -pd, -t1
Traceback (most recent call last):
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/_pydevd_bundle/pydevd_comm.py", line 1466, in do_it
    result = pydevd_console_integration.console_exec(self.thread_id, self.frame_id, self.expression, dbg)
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/_pydevd_bundle/pydevd_console_integration.py", line 203, in console_exec
    Exec(code, updated_globals, frame.f_locals)
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/_pydevd_bundle/pydevd_exec2.py", line 3, in Exec
    exec(exp, global_vars, local_vars)
  File "<input>", line 1, in <module>
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 1766, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 2001, in _parse_known_args
    ', '.join(required_actions))
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 2393, in error
    self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 2380, in exit
    _sys.exit(status)
SystemExit: 2
~~~
investigating...

Reply:
fixed in 19f18a69ed32f79d3dcdda2931c58af91d61a7cf